eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/AggregateBasedStorageEngineTestSuite.java

```diff
@@ -165,6 +192,38 @@ void streamingAfterLastPositionReturnsEmptyStream() {
         }
     }
 
+    @Test
+    void sourcingEventsReturnsMatchingAggregateEvent() {
+        AppendCondition appendCondition = AppendCondition.withCriteria(TEST_AGGREGATE_CRITERIA);
+        testSubject.appendEvents(appendCondition,
```

Comment:
This test now validates that the sole entry is also returned. So, the entire given is the entire expected. That makes it tough to state that "matching" is actually performed. Perhaps it would be good to add several events, including events that don't match the sourcing condition.

Reply:
We already have such tests (eg. `sourcingEventsReturnsMatchingAggregateEvents`), which you described.
I just add this one as a first step, while I started implementation I didn't have simple test to fulfill, so I added this one.